### ISSUE TITLE
Support for nesting a schema within itself

### ISSUE THREAD

**Issue by nryang** [sv-tools/marshmallow-objects#79]
If you define a Marshmallow object that has a field that references itself, you get an error.

Example:
```py
class Node(marshmallow.Model):
    nodes = marshmallow.NestedModel(lambda: Node(exclude=("nodes",)))
```

Error:
```
site-packages/marshmallow_objects/models.py", line 86, in __init__
    super(NestedModel, self).__init__(nested.__schema_class__, **kwargs)
AttributeError: 'function' object has no attribute '__schema_class__'
```

**Comment by SVilgelm** [comment]
Hi @nryang thank you, I will check what I can do to fix your issue.

**Comment by SVilgelm** [comment]
Hi, unfortunately the way how it works in marshmallow doesn't work with marshmallow_objects.
Bu I solved this problem. With a #83 you can use a text name of a class to achieve a nesting schema within itself:

```python
class Person(marshmallow.Model):
    name = marshmallow.fields.String()
    friend = marshmallow.NestedModel("Person")

Person.load({"name": "John Doe", "friend": {"name": "Jane Doe"}})
```

**Comment by SVilgelm** [comment]
Released in 2.2.0: https://pypi.org/project/marshmallow-objects/2.2.0/
